src/Foundatio/Utility/ScheduledTimer.cs

```diff
@@ -25,8 +27,8 @@ public class ScheduledTimer : IDisposable {
             _timerCallback = timerCallback;
             _minimumInterval = minimumIntervalTime ?? TimeSpan.Zero;
 
-            int dueTimeMs = dueTime.HasValue ? (int)dueTime.Value.TotalMilliseconds : Timeout.Infinite;
-            _timer = new Timer(s => RunCallbackAsync().GetAwaiter().GetResult(), null, dueTimeMs, Timeout.Infinite);
+            if (dueTime != null)
+                _timer = SystemClock.Instance.Schedule(dueTime.Value, async () => await RunCallbackAsync());
```

Comment:
It doesn't look like the Schedule method has any overloads that take tasks and the code seems to be calling the overload that just takes an Action. Wouldn't that mean that this would be an async void call?